ui/src/app/pages/analysis/runtime-panel.tsx

```diff
@@ -480,11 +484,116 @@ const CostEstimator = ({
   </FlexRow>;
 };
 
+const CreatePanel = ({creatorFreeCreditsRemaining, preset, profile, setPanelContent, workspace}) => {
+  const {
+    displayName,
+    runtimeTemplate: {gceConfig = null, dataprocConfig = null}
+  } = preset;
+  const computeType = dataprocConfig ? ComputeType.Dataproc : ComputeType.Standard;
+  const masterMachineType = computeType === ComputeType.Dataproc ? dataprocConfig.masterMachineType : gceConfig.machineType;
+  const masterMachine = findMachineByName(masterMachineType);
+  const masterDiskSize = computeType === ComputeType.Dataproc ? dataprocConfig.masterDiskSize : gceConfig.diskSize;
+  const workerMachineType = computeType === ComputeType.Dataproc ? dataprocConfig.workerMachineType : null;
+  const workerMachine = workerMachineType ? findMachineByName(workerMachineType) : null;
+  const workerDiskSize = computeType === ComputeType.Dataproc ? dataprocConfig.workerDiskSize : null;
+
+  return <div style={styles.controlSection}>
+    <CostEstimator
+        freeCreditsRemaining={creatorFreeCreditsRemaining}
+        profile={profile}
+        runtimeParameters={{
+          computeType,
+          diskSize: gceConfig && gceConfig.diskSize,
+          machineType: gceConfig && gceConfig.machineType,
+          dataprocConfig: dataprocConfig
+        }}
+        runtimeChanged={false}
+        workspace={workspace}
+    />
+    <FlexRow style={{justifyContent: 'space-between', alignItems: 'center'}}>
+      <h3 style={{...styles.sectionHeader, ...styles.bold}}>Recommended Environment for {displayName}</h3>
+      <Button
+          type='secondarySmall'
+          onClick={() => setPanelContent(PanelContent.Customize)}
+          aria-label='Customize'
+      >
+        Customize
+      </Button>
+    </FlexRow>
+    <label htmlFor='compute-resources' style={{...styles.bold, marginTop: '1rem'}}>Compute Resources</label>
+    <div id='compute-resources'>- Default: compute size of
+      <b> {masterMachine.cpu} CPUs</b>,
+      <b> {masterMachine.memory} GB memory</b>, and a
+      <b> {masterDiskSize} GB disk</b>
+    </div>
+    {computeType === ComputeType.Dataproc && <Fragment>
+      <label htmlFor='worker-configuration' style={{...styles.bold, marginTop: '1rem'}}>Worker Configuration</label>
+      <div id='worker-configuration'>- Default:
+        <b> {dataprocConfig.numberOfWorkers} worker(s) </b>
+        {
+          dataprocConfig.numberOfPreemptibleWorkers > 0 &&
+          <b>and {dataprocConfig.numberOfPreemptibleWorkers}  preemptible worker(s) </b>
+        }
+        each with compute size of <b>{workerMachine.cpu} CPUs</b>,
+        <b> {workerMachine.memory} GB memory</b>, and a
+        <b> {workerDiskSize} GB disk</b>
+      </div>
+    </Fragment>}
+  </div>;
+};
+
+const UpdateRuntimeButton = ({
+   closePanel,
+   runtimeExists,
```

Comment:
nit: I feel it might be a cleaner separation to replace these constituent properties with `disabled` and `label` (disabled could default to `false`). This way you don't leave any opportunity for bugs where the button is potentially grayed out on the creation panel (create panel would just be a simple hardcoded `'Create'`), and it pushed most of the hairy business logic into the customize panel where it belongs.

Reply:
Good call, will do.